pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -48,13 +50,30 @@ def __init__(self) -> None:
         self.running_loss: TensorRunningAccum = TensorRunningAccum(window_length=20)
         self.batch_idx: int = 0
         self.split_idx: Optional[int] = None
-        self._warning_cache: WarningCache = WarningCache()
+        self.progress = BatchProgress()
+        self.optim_progress = OptimizationProgress()
```

Comment:
great!
Let's say we would break out the optimizer loop into it's own loop class. 
We would then simply move this property to that optimizer loop?

Reply:
Depends on the responsabilities of the optimizer loop but likely yes.

Do you have something more specific in mind?

Reply:
just wanted to confirm we have the flexibility available. 
what we would track as progress in such an optimizer loop is not yet so clear but we will see :)